src/Bridges/lazy_bridge_optimizer.jl

```diff
@@ -41,6 +41,9 @@ mutable struct LazyBridgeOptimizer{OT<:MOI.ModelLike} <: AbstractBridgeOptimizer
     objective_bridge_types::Vector{Any}
     objective_node::OrderedDict{Tuple{DataType}, ObjectiveNode}
     objective_types::Vector{Tuple{DataType}}
+    # Cache for (F, S) -> BridgeType. Avoids having to look up
+    # `concrete_bridge_type`, which is slow.
```

Comment:
I would still add something like "method look up for the function `concrete_bridge_type` at run-time". It's quite subtle, even for an advanced Julia users so more explanation would be helpful